### DISCUSSION TITLE
Pushbutton Only Resets on Battery

### DISCUSSION THREAD

**Discussion by GRBlake488** [alanbjohnston/CubeSatSim#257] [General]
When the Cube Sat Simulator is only on battery, the pushbutton, when depressed, will only RESET the Pi. If I power the stack from the Pi USB port, then it will allow me to press and hold for mode selection.

Anyone else have this problem?

George Blake

**Comment by alanbjohnston** [comment]
Hi George,

I've never seen this before.  Is your battery fully charged?  The Pi will reboot if the 5V power hiccups or drops suddenly.  After it reboots, you could inspect the log files by logging into your Pi and typing:

CubeSatSim/log

If you copy and paste the last 50 or so lines that are displayed here, I might be able to see what is going on.

You can also try re-installing the pushbutton code by typing this:

pi-power-button/script/install

After you do this, you will also be able to see what happens when you press the pushbutton until the next reboot of your Pi.  You will see a short text message displayed on the console when the pushbutton is released.

Good luck!

73,
Alan

**Comment by GRBlake488** [comment]
Alan,
I will check using a DMM again. It was plugged into the USB Power port right up until I tried it. After I confirmed it was working with the Pi port and not on battery I plugged it in again. I then tested it the same way again just to confirm.
I will check to see if there is a dip in the voltage when I unplug it.
I can send you the files later tonight.
73's.
George Blake


    On Thursday, January 5, 2023, 01:33:58 AM CST, alanbjohnston ***@***.***> wrote:  
 
 


Hi George,

I've never seen this before. Is your battery fully charged? The Pi will reboot if the 5V power hiccups or drops suddenly. After it reboots, you could inspect the log files by logging into your Pi and typing:

CubeSatSim/log

If you copy and paste the last 50 or so lines that are displayed here, I might be able to see what is going on.

You can also try re-installing the pushbutton code by typing this:

pi-power-button/script/install

After you do this, you will also be able to see what happens when you press the pushbutton until the next reboot of your Pi. You will see a short text message displayed on the console when the pushbutton is released.

Good luck!

73,
Alan

**Comment by GRBlake488** [comment]
Alan,
I found the problem. Short between the headsets and trippping over my own coax.  :P
The holes in the PCB are too small for #22 AWG wire. I don't like really small holes in boards and I like to have higher current potential. I tried to open the holes for the battery with a small PCB Awl, but wrecked the annular holes.

I soldered the ground / return wire to the pushbutton switch because I could pick up two pins to solder to. Unfortunately, I got the N.C. contact instead of the COM contacts in the center that go to ground. So, when I pushed the button, it opened the ground side of the batteries. When plugged into USB, the ground return was through the USB connection.
Thanks for the tip on the PI Tx line. I will use that.
What about SPI code? I saw it in the original code. Is it still in this iteration? That should run the 2.4 GHz Transceiver.

73's.
George Blake






    On Thursday, January 5, 2023 at 01:33:58 AM CST, alanbjohnston ***@***.***> wrote:  
 
 


Hi George,

I've never seen this before. Is your battery fully charged? The Pi will reboot if the 5V power hiccups or drops suddenly. After it reboots, you could inspect the log files by logging into your Pi and typing:

CubeSatSim/log

If you copy and paste the last 50 or so lines that are displayed here, I might be able to see what is going on.

You can also try re-installing the pushbutton code by typing this:

pi-power-button/script/install

After you do this, you will also be able to see what happens when you press the pushbutton until the next reboot of your Pi. You will see a short text message displayed on the console when the pushbutton is released.

Good luck!

73,
Alan
